engine/notifier_test.go

```diff
@@ -141,24 +149,25 @@ func TestNotifier_AllWorkProcessed(t *testing.T) {
 
 		// 5 routines consuming work
 		for i := 0; i < 5; i++ {
+			startingRoutines.Add(1)
 			go func() {
+				startingRoutines.Done()
+				start.Wait()
 				for consumedWork.Load() < totalWork {
 					<-notifier.Channel()
-				L:
 					for {
 						select {
 						case <-pendingWorkQueue:
 							consumedWork.Inc()
 						default:
-							break L
+							break
```

Comment:
Did we not want to keep the label [as discussed here](https://github.com/onflow/flow-go/pull/1037#issuecomment-887896284)?

Reply:
thanks for the reminder. Absolutely this should break the `for` loop